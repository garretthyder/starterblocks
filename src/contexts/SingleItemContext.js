import { createContext } from '@wordpress/element';
const SingleItemContext = createContext({});
export const SingleItemProvider = SingleItemContext.Provider;
export const SingleItemConsumer = SingleItemContext.Consumer;
export default SingleItemContext;
