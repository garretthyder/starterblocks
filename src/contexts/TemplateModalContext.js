import { createContext } from '@wordpress/element';
const TemplateModalContext = createContext({});
export const TemplateModalProvider = TemplateModalContext.Provider;
export const TemplateModalConsumer = TemplateModalContext.Consumer;
export default TemplateModalContext;
