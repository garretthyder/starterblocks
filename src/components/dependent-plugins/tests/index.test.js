import renderer from 'react-test-renderer';
import {mount} from 'enzyme';
import DependentPlugins from '..';
import {SingleItemProvider} from '../../../contexts/SingleItemContext';

let singleMock = {
    data: {ID: 1, blocks: {}},
    showDependencyBlock: true
};

const WrappedDependentPlugins = (props) => {
    const {singleValue} = props;
    return (
        <SingleItemProvider value={{...singleMock, ...singleValue}}>
            <DependentPlugins />
        </SingleItemProvider>
    );
}

describe('Dependent Plugins part within Button Group component', () => {
    it('1. renders correctly: snapshot testing', () => {
        const component = renderer.create(
            <WrappedDependentPlugins />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('2. Testing props', () => {
        it('renders nothing when showDependencyBlock of SingleItemProvider is false', () => {
            const component = renderer.create(
                <WrappedDependentPlugins singleValue={{showDependencyBlock: false}} />
            );
            expect(component.root).toBe(null)
        });
    });

});
