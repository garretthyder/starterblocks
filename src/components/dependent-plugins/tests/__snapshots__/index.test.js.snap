// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dependent Plugins part within Button Group component 1. renders correctly: snapshot testing 1`] = `
<div
  className="starterblocks-button-display-dependencies"
/>
`;
